roachtest: add cdc-filtering roachtest

nicktrav opened the pull request:
Verify the end-to-end correctness of the Session Based CDC Filtering feature via a roachtest.

The test sets up a table of (id, revision) tuples, with a changefeed emitting events into a file based sink on n1. Certain updates to the table are made with the `disable_changefeed_replication` session variable set to `true`, preventing the events from being emitted on the changefeeed.

The changefeed events are then read from the file based sink, converted into a canonical representation, before being compared to the expected sequence of events.

Touches #114074.

Epic: CRDB-13169.

Release note: None.

nicktrav committed:
roachtest: add cdc-filtering roachtest

Verify the end-to-end correctness of the Session Based CDC Filtering
feature via a roachtest.

The test sets up a table of (id, revision) tuples, with a changefeed
emitting events into a file based sink on n1. Certain updates to the
table are made with the `disable_changefeed_replication` session
variable set to `true`, preventing the events from being emitted on the
changefeeed.

The changefeed events are then read from the file based sink, converted
into a canonical representation, before being compared to the expected
sequence of events.

Touches #114074.

Release note: None.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/116521)
<!-- Reviewable:end -->

nicktrav commented:
Blocked on #116065. This works locally, when rebased on top of those commits.

miretskiy reviewed (approved):
nice

miraradeva reviewed (approved):
Thank you for testing this end-to-end! 
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

If you want to increase the test coverage, consider adding the `diff` option as well and asserting that filtered out writes appear as previous values.

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/116521#-:-Nli5U3y4pl-YzHk4d3Q:b-6ctf8n)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @andyyang890, @DarrylWong, and @renatolabs)



<!-- Sent from Reviewable.io -->

nicktrav commented:
> consider adding the diff option as well and asserting that filtered out writes appear as previous values.

Good call. Updated to use `diff`.

I also rebased, so this should be good to go now.

TFTRs!

nicktrav commented:
bors r+
